### PR TITLE
Poll ranking crud

### PR DESCRIPTION
**Pull request by afnantuffaha**
# Description

Implements ranking polls where users rank all options, with results aggregated using Borda count scoring to determine the top 3.

- `POST /trips/:id/rank-polls` - Create rank poll

- `PATCH /trips/:id/rank-polls/:pollId` - Update poll metadata (question, deadline only)

- `POST /trips/:id/rank-polls/:pollId/options` - Add option (only if no rankings yet)

- `DELETE /trips/:id/rank-polls/:pollId/options/:optionId` - Delete option (only if no rankings yet)

- `POST /trips/:id/rank-polls/:pollId/rank` - Submit or replace full ranking

- `GET /trips/:id/rank-polls/:pollId` - Get results (aggregate ranks + top 3)

- `DELETE /trips/:id/rank-polls/:pollId` - Delete poll

- `GET /trips/:id/rank-polls/:pollId/voters` - Get who voted and who didn't vote

Real-time updates are published when rankings are submitted.

## How has this been tested?

Tested with integration tests for all endpoints.

## Checklist

* [x]  I have self-reviewed my code for readability, maintainability, performance, and added comments/documentation where necessary.
* [x] New and existing tests pass locally with my changes.
* [x] I have followed the project's coding standards and best practices.

- Users can create full-ranking (Borda) polls, submit/replace complete rankings, and view aggregated results (top 3, per-option scores, avg rank, vote counts); submissions publish realtime updates.
- Poll creators can edit question and deadline (subject to deadline/ownership rules); options may be added/removed only before any rankings exist; strict validation enforces full contiguous rankings and option-count parity.
- New endpoints expose poll lifecycle, option management, ranking submission, aggregated results, and voter status (who has/hasn't voted); comprehensive integration tests and a frontend test screen exercise the flow.

Features
- New REST endpoints:
  - POST /trips/:id/rank-polls — create rank poll
  - GET /trips/:id/rank-polls/:pollId — aggregated results (top 3, options with scores, user ranking)
  - PATCH /trips/:id/rank-polls/:pollId — update question & deadline (creator-only constraints)
  - DELETE /trips/:id/rank-polls/:pollId — delete poll (cascades rankings)
  - POST /trips/:id/rank-polls/:pollId/options — add option (only if no rankings)
  - DELETE /trips/:id/rank-polls/:pollId/options/:optionId — delete option (only if no rankings, min options enforced)
  - POST /trips/:id/rank-polls/:pollId/rank — submit or replace a full ranking (validated)
  - GET /trips/:id/rank-polls/:pollId/voters — list trip members with voting status
- Borda aggregation: score computed as (num_options - rank_position + 1); responses include OptionWithScore (score, avg_rank, vote_count) and top_3.

Improvements
- Robust validation: 3–15 options, ranks must cover 1..N with no gaps/duplicates, option IDs uniqueness, rank count == option count.
- Access, ownership & deadline enforcement: membership required; creator/admin checks for sensitive ops; blocks modifications per business rules.
- Realtime: publishes "poll.ranking_submitted" (and poll.created/updated/deleted) events.
- API & docs: OpenAPI/Swagger expanded with poll/ranking models and endpoints.
- Frontend: test screen added to exercise the new rank-poll API.

Infrastructure
- DB migration: new poll_rankings table (poll_id, user_id, option_id, rank_position) with composite PK, FKs, and indexes.
- New service/repo/controller/router wiring:
  - RankPollService (interface + impl) — create/update/add/delete options, submit rankings, get results/voters, delete polls.
  - PollRankingRepository — transactional submit/replace rankings, aggregated Borda results, voter status queries.
  - RankPollController + route group /trips/:tripID/rank-polls with TripMemberRequired middleware.
  - Realtime registry allowed topic "poll.ranking_submitted".
- Tests: comprehensive integration tests covering lifecycle, validation, aggregation, authorization, and edge cases.

Sequence (submit ranking)
Client -> POST /trips/:tripID/rank-polls/:pollId/rank
Controller -> validate auth/params/body
Service -> validate business rules; call Repository.SubmitRanking (transaction: delete existing user rankings → insert new rows)
Repository -> persist poll_rankings; compute aggregates
Service -> publish "poll.ranking_submitted" with aggregated results
Controller -> return 200/201

Author Contribution (lines)
| Component | Added | Removed | Net |
| API Documentation (docs.go, swagger.json, swagger.yaml) | 4,295 | 50 | +4,245 |
| Integration Tests | 1,359 | 0 | +1,359 |
| Frontend (test screen + nav) | 620 | 0 | +620 |
| Service Layer | 499 | 0 | +499 |
| HTTP Controller | 346 | 0 | +346 |
| Repository (poll ranking) | 175 | 0 | +175 |
| Data Models | 73 | 0 | +73 |
| Router & wiring (routes, repo init) | 51 | 16 | +35 |
| Database Migration | 25 | 0 | +25 |
| Real-time Registry | 1 | 0 | +1 |
| Misc (small docs/tests adjustments) | 2 | 2 | 0 |
| Totals | 7,446 | 68 | +7,378 |

### CODE REVIEW

File: backend/internal/controllers/poll_ranking.go [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+		return err
+	}
+
+	poll, err := rc.rankPollService.CreateRankPoll(c.Context(), tripID, userID, req)
+	if err != nil {
+		return err
+	}
+
+	return c.Status(http.StatusCreated).JSON(poll)
+}
```
[the remark this answers is not in the document]
Comment (in-mai-space) [reply backend/internal/controllers/poll_ranking.go]: good comment, let's take a look here @afnantuffaha

File: backend/internal/repository/poll_ranking.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+		EntityType  *string           `bun:"entity_type"`
+		EntityID    *uuid.UUID        `bun:"entity_id"`
+		BordaScore  int               `bun:"borda_score"`
+		AverageRank float64           `bun:"average_rank"`
+		VoteCount   int               `bun:"vote_count"`
+	}
+
+	var rows []resultRow
+
+	err := r.db.NewSelect().
```
[the remark this answers is not in the document]
Comment (afnantuffaha) [reply backend/internal/repository/poll_ranking.go]: i ran it and the performance looks good!
